wal/wal.go

```diff
@@ -172,11 +172,8 @@ func openAtIndex(dirpath string, snap walpb.Snapshot, all bool) (*WAL, error) {
 		}
 		err = l.TryLock()
 		if err != nil {
-			if all {
+			if write {
 				return nil, err
-			} else {
```

Comment:
After discussion OOL, the TryLock in read mode is a best-effort one and we don't care about whether it is successful.